gamemodes/terrortown/gamemode/server/sv_weaponry.lua

```diff
@@ -627,7 +634,53 @@ function GM:WeaponEquip(wep, ply)
 
 	if IsValid(ply) and wep.Kind then
 		AddWeaponToInventoryAndNotifyClient(ply, wep)
+
+		-- there is a glitch that picking up a weapon does not refresh the weapon cache on
+		-- the client. Therefore the client has to be notified to updated its cache
+		net.Start("ttt2_switch_weapon_update_cache")
+		net.Send(ply)
 	end
+
+	-- handle all this stuff in the next frame since the owner is not yet valid
+	timer.Simple(0, function()
```

Comment:
please localize this function

Reply:
I need `wep` and `ply`
all I could do would be to create a new function and call it inside this.

Reply:
you can localize this function right before